src/ls_server.rs

```diff
@@ -251,34 +252,36 @@ impl LsService {
     }
 
     fn init(&self, id: usize, init: InitializeParams) {
-        let result = InitializeCapabilities {
+        let result = InitializeResult {
             capabilities: ServerCapabilities {
-                textDocumentSync: DocumentSyncKind::Incremental as usize,
-                hoverProvider: true,
-                completionProvider: CompletionOptions {
-                    resolveProvider: true,
-                    triggerCharacters: vec![".".to_string()],
-                },
+                text_document_sync: Some(TextDocumentSyncKind::Incremental),
```

Comment:
It's just the way the LSP protocol is modelled. Arguably the protocol itself could be simplified in this regard, but otherwise I think we should model the protocol types precisely (even if it has redundancies). That is not to say we couldn't create a shortcut ctor function, or even an intermediate type (say, to `ServerCapabilities`), that would not have the Options. I'll give that a try.